woocommerce/google-listings-and-ads
# Feature/1036 submit verification code keyboard

### Changes proposed in this Pull Request:

Adding a feature for allowing the user to fastly submit the verification code by pressing "Enter"

Closes #1036 


### Screenshots:

![ezgif-6-fa17f0d0222b](https://user-images.githubusercontent.com/5908855/142432841-3853bcf3-e06c-40b9-8cac-a87f3102f053.gif)

<img width="1218" alt="Screenshot 2021-11-18 at 15 27 55" src="https://user-images.githubusercontent.com/5908855/142434170-1a62769d-5458-47ec-bccc-eda54020591e.png">

### Detailed test instructions:

1. Enter into the SetupMerchant onboarding page
2. Go to step 4 (Confirmation)
3. Add your country code and phone
4. Click on Verify Phone. Number
5a. Clicking enter without all the digits filled doesn't submit the code
5b. Clicking enter with all the digits filled submit the verification code

Up to you, (since it's the same component will work the same), but there is also another way to access the component in:

1. Go to GLA settings page once setup is completed
2. Click "Edit" in the phone number card
3. Repeat steps 3-4-5  

Run the tests using `npm run test-unit` 

### Technical summary

> Add - Test suite for Validation Code Control 
> Add - Keyboard accesibility 

> Update - Minor refactoring in validation code Control and Verify Phone Number Content.

- Local function `handleVerifyClick` is now `handleVerification` since we are calling it now unbutton click and by keyboard enter press.
- Local Computed variable `disableVerify`is renamed to `disabledVerify` when I looked at the code I thought was an imperative order "Disable the verify" but is actually a state check "Is verify disabled?" 

### Changelog entry

> Added test suite for Validation Code Control
> Improve accessibility for Validation Code Control
